cmake/jsoncpp.cmake

```diff
@@ -10,6 +10,7 @@ include(GNUInstallDirs)
 set(prefix "${CMAKE_BINARY_DIR}/deps")
 set(JSONCPP_LIBRARY "${prefix}/${CMAKE_INSTALL_LIBDIR}/${CMAKE_STATIC_LIBRARY_PREFIX}jsoncpp${CMAKE_STATIC_LIBRARY_SUFFIX}")
 set(JSONCPP_INCLUDE_DIR "${prefix}/include")
+add_compile_options(-isystem ${JSONCPP_INCLUDE_DIR})
```

Comment:
So this declares it as a system header and it won't enforce some restrictions on it? Maybe it is a good idea to do this nonetheless?

What do you think @chfast ?

Reply:
ah yeah.. I also saw that the stack for enabled address sanitizers is also not large enough for osx.. i increased it to 32mb.. what is actually the default stacksize on linux.. 16mb?

Reply:
@chfast nice :) thx

Reply:
Do you want to create a separate PR for this?

Reply:
see PR #4143